pkg/resources/saml2_integration.go

```diff
@@ -0,0 +1,891 @@
+				if err = d.Set("saml2_force_authn", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["saml2_snowflake_acs_url"]; !v.IsNull() {
+				if err = d.Set("saml2_snowflake_acs_url", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["saml2_snowflake_issuer_url"]; !v.IsNull() {
+				if err = d.Set("saml2_snowflake_issuer_url", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+			if v := d.GetRawConfig().AsValueMap()["saml2_snowflake_x509_cert"]; !v.IsNull() {
+				if err = d.Set("saml2_snowflake_x509_cert", v.AsString()); err != nil {
+					return diag.FromErr(err)
+				}
+			}
+		}
+
+		if err = d.Set(showOutputAttributeName, []map[string]any{schemas.SecurityIntegrationToSchema(integration)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err = d.Set(describeOutputAttributeName, []map[string]any{schemas.DescribeSaml2IntegrationToSchema(integrationProperties)}); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+}
+
+func UpdateContextSAML2Integration(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	set, unset := sdk.NewSaml2IntegrationSetRequest(), sdk.NewSaml2IntegrationUnsetRequest()
+
+	if d.HasChange("enabled") {
+		if v := d.Get("enabled").(string); v != "unknown" {
+			parsed, err := strconv.ParseBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithEnabled(parsed)
+		} else {
+			// UNSET is not implemented
+			set.WithEnabled(true)
+		}
+	}
+
+	if d.HasChange("saml2_issuer") {
+		set.WithSaml2Issuer(d.Get("saml2_issuer").(string))
+	}
+
+	if d.HasChange("saml2_sso_url") {
+		set.WithSaml2SsoUrl(d.Get("saml2_sso_url").(string))
+	}
+
+	if d.HasChange("saml2_provider") {
+		valueRaw := d.Get("saml2_provider").(string)
+		value, err := sdk.ToSaml2SecurityIntegrationSaml2ProviderOption(valueRaw)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		set.WithSaml2Provider(value)
+	}
+
+	if d.HasChange("saml2_x509_cert") {
+		set.WithSaml2X509Cert(d.Get("saml2_x509_cert").(string))
+	}
+
+	if d.HasChange("saml2_sp_initiated_login_page_label") {
+		// UNSET is not implemented (conditional ForceNew on unset)
+		set.WithSaml2SpInitiatedLoginPageLabel(d.Get("saml2_sp_initiated_login_page_label").(string))
+	}
+
+	if d.HasChange("saml2_enable_sp_initiated") {
+		if v := d.Get("saml2_enable_sp_initiated").(string); v != "unknown" {
+			parsed, err := strconv.ParseBool(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithSaml2EnableSpInitiated(parsed)
+		} else {
+			// UNSET is not implemented
+			set.WithSaml2EnableSpInitiated(false)
+		}
+	}
+
+	if d.HasChange("saml2_snowflake_x509_cert") {
+		// UNSET is not implemented (conditional ForceNew on unset)
```

Comment:
it would be nice to have one issue for all unsets not working for security integrations (so we have context and so that we can come back)

Reply:
Added todos